File: frontend/bff/src/application/organizations/commands/update-company-profile.command.ts
import type { RequestContext } from '../../../cross-cutting/context/request-context.js';
import type { Sector } from '../../../cross-cutting/enums/sector.js';

export interface UpdateCompanyProfileCommand {
  context: RequestContext;
  corporateName: string;
  taxId: string | null;
  headquarters: string | null;
  group: string | null;
  sector: Sector;
  /** Mês de início do ano-safra (1–12). */
  cropYearStartMonth: number;
  /** AA/AA */
  activeCrop: string | null;
}

File: backend/core-service/src/Fix.Application/UseCases/Organizations/Commands/UpdateCompanyProfileCommand.cs
using Fix.Application.Abstractions.Authorization;
using Fix.Application.Abstractions.Context;
using Fix.Application.Abstractions.Messaging;
using Fix.Application.Organizations.Dtos;
using Fix.Domain.AggregateRoots.Organizations;
using Fix.Domain.AggregateRoots.Roles;

namespace Fix.Application.Organizations.Commands;

[RequireRole(RoleCodes.EditOrganization)]
public sealed record UpdateCompanyProfileCommand(
    Guid UserId,
    Guid OrganizationId,
    string CorporateName,
    string? TaxId,
    string? Headquarters,
    string? Group,
    Sector Sector,
    int CropYearStartMonth,
    string? ActiveCrop)
    : ICommand<OrganizationSetupDto>, IOrganizationRequest;

